bokeh/util/notebook.py

```diff
@@ -0,0 +1,88 @@
+""" Functions useful for loading Bokeh code and data in IPython notebooks.
+
+"""
+from __future__ import absolute_import
+
+_notebook_loaded = None
+
+def load_notebook(resources=None, verbose=False, hide_banner=False):
+    """ Prepare the IPython notebook for displaying Bokeh plots.
+
+    Args:
+        resources (Resource, optional) :
+            how and where to load BokehJS from
+
+        verbose (bool, optional) :
+            whether to report detailed settings (default: False)
+
+        hide_banner (bool, optional):
+            whether to hide the Bokeh banner (default: False)
+
+    .. warning::
+        Clearing the output cell containing the published BokehJS
+        resources HTML code may cause Bokeh CSS styling to be removed.
+
+    Returns:
+        None
+
+    """
+    global _notebook_loaded
+
+    from .. import __version__
+    from ..resources import INLINE
+    from ..templates import NOTEBOOK_LOAD, RESOURCES
```

Comment:
usually prefer to group all important at the beginning of the file for better organization, unless there's a benefit in scoping imports in a function...